Source/WebCore/page/scrolling/mac/ScrollerMac.mm

```diff
@@ -370,6 +370,11 @@ - (void)invalidate
     return FloatPoint { [m_hostLayer convertPoint:point fromLayer:[m_hostLayer superlayer]] };
 }
 
+void ScrollerMac::updateScrollbarStyle()
+{
+    m_scrollerImp = [NSScrollerImp scrollerImpWithStyle:nsScrollerStyle(m_pair.scrollbarStyle()) controlSize:NSControlSizeRegular horizontal:false replacingScrollerImp:takeScrollerImp().get()];
```

Comment:
Do we need to update the `m_scrollerImp` with any of the current state (`doubleValue` etc), or will it get updated before the next paint?

Reply:
We provide the previous NSScrollerImp so appkit carries over that state to the new NSScrollerImp.

Reply:
Also what happens to the entry in the `ScrollbarToScrollerImpMap`?

Reply:
These NSScrollerImps are not registered in that map. That is only used when we create Scrollbar objects in the WebProcess.